### PR TITLE
Implement MiniscriptKey for secp256k1 PublicKey

### LINKED ISSUE #326: Consider impl MiniscriptKey for secp256k1::PublicKey
`secp256k1::PublicKey` is the only type that can be used to derive both `XOnlyPublicKey` and `bitcoin::PublicKey`. This pattern is useful a lot in downstream wallets, when they do not know which key type will be held by descriptors.

### PR DESCRIPTION
**Pull request by dr-orlovsky**
Closes #326

### CODE REVIEW

**Review by sanket1729 — COMMENTED** [review]
Would like to see the doc-comment fixed because it says the opposite of what the code is doing.

File: src/lib.rs [hunk 1]
```diff
@@ -155,6 +156,23 @@ pub trait MiniscriptKey: Clone + Eq + Ord + fmt::Debug + fmt::Display + hash::Ha
     fn to_pubkeyhash(&self) -> Self::Hash;
 }
 
+impl MiniscriptKey for bitcoin::secp256k1::PublicKey {
+    /// `is_uncompressed` always returns `true`
```
[the remark this answers is not in the document]
Comment (dr-orlovsky) [reply src/lib.rs]: Sorry, `false` for negative statement (_un_compressed) broke my natural language processor.

**Review by apoelstra — APPROVED** [review]
ACK 0888fe9ee625c9b7e8a8aff8d9bf4aa75f88b248

**Review by sanket1729 — APPROVED** [review]
ACK 0888fe9ee625c9b7e8a8aff8d9bf4aa75f88b248

**Pull request merged**